addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -98,31 +98,30 @@ def _check_move_configuration(self, move):
                 error_message.append(_("- Transport document number and date is required when Transportation Mode is Rail,Air or Ship"))
         if error_message:
             error_message.insert(0, _("The following information are missing on the invoice (see eWayBill tab):"))
-        goods_lines = move.invoice_line_ids.filtered(lambda line: not (line.display_type in ('line_section', 'line_note', 'rounding') or line.product_id.type == "service"))
-        if not goods_lines:
+        empty_hsn_indices = []
+        empty_tax_indices = []
+        goods_lines = move.invoice_line_ids.filtered(lambda line: not (line.display_type in ('line_section', 'line_note', 'rounding'))).sorted(lambda l: (l.sequence, l.id))
+        for line_number, line in enumerate(goods_lines, start=1):
+            hsn_code = self._l10n_in_edi_extract_digits(line.l10n_in_hsn_code)
+            if not hsn_code:
+                empty_hsn_indices.append(str(line_number))
+            if not line.tax_ids:
+                empty_tax_indices.append(str(line_number))
+        if all(line.product_id.type == "service" for line in goods_lines if line.product_id):
             error_message.append(_('You need at least one product having "Product Type" as stockable or consumable.'))
+        if empty_hsn_indices:
+            error_message.append(_("Line number %s does not have HSN/SAC code.", ','.join(empty_hsn_indices)))
```

Comment:
```suggestion
            error_message.append(_("Line number %(number_list)s does not have HSN/SAC code.", number_list=', '.join(empty_hsn_indices)))
```
Gives translator context + spaces are important
Also, as it's a l10n module, you should reexport the .pot